modules/db/src/main/scala/gem/dao/SmartGcalDao.scala

```diff
@@ -317,5 +323,69 @@ object SmartGcalDao {
 
     val bulkInsertGmosSouth: Update[GmosSouthRow] =
       Update[GmosSouthRow](bulkInsertGmosSql("smart_gmos_south"))
+
+    // -----------------------------------------------------------------------
+    // GNIRS
+    // -----------------------------------------------------------------------
+
+    val createIndexGnirs: Update0 =
+      sql"""
+        CREATE INDEX IF NOT EXISTS smart_gnirs_index ON smart_gnirs
+          (camera, disperser, filter, fpu_slit, fpu_other, prism, well_depth)
```

Comment:
From the [postgres doc](https://www.postgresql.org/docs/9.5/static/indexes-multicolumn.html), 

> Multicolumn indexes should be used sparingly. In most situations, an index on a single column is sufficient and saves space and time. Indexes with more than three columns are unlikely to be helpful unless the usage of the table is extremely stylized.

For GMOS, which is huge, having 3-column indices did help query performance.  GNIRS is comparatively tiny and I think we'd be fine with picking just a couple that fairly well evenly divide the table.